packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.d.ts

```diff
@@ -195,3 +195,10 @@ export default function PgIntrospectionPlugin(
   builder: SchemaBuilder,
   options: Options
 ): Promise<void> | void;
+
+import "graphile-build";
+declare module "graphile-build" {
+  interface BuildPropertiesIntroducedIn_build {
+    pgIntrospectionResultsByKind: PgIntrospectionResultsByKind;
+  }
+}
```

Comment:
This is a good example on how most plugins would be using it to extend Build.

In the build phase, `pgIntrospectionResultsByKind` will be `undefined | PgIntrospectionResultsByKind` - in all later phases, it will be `PgIntrospectionResultsByKind`. 
In earlier phases (okay, there's nothing earlier thatn build, but you get the gist), the property wouldn't be present.